content/api/node.md

```diff
@@ -1,7 +1,301 @@
 ---
 title: Node.js API
+contributors:
+  - sallar
 ---
-> webpack()
-> run()
-> compile()
-> watch()
+## Overview
+
+webpack provides a Node.js API which can be used directly in Node.js runtime.
+
+The Node.js API is useful in scenarios in which you need to customize the build or development process since all the reporting and error handling must be done manually and webpack only does the compiling part.
+
+## Installation
+
+To start using webpack Node.js API, first install webpack if you haven’t yet:
+
+```
+npm install webpack --save-dev
+```
+
+Then require the webpack module in your Node.js script:
+
+``` js
+const webpack = require("webpack");
+
+// Or if you prefer ES6:
+import webpack from "webpack";
+```
+
+## `webpack()`
+
+The imported `webpack` function is fed a webpack [Configuration Object](/configuration/) and runs the webpack compiler if a callback function is provided:
+
+``` js-with-links
+const webpack = require("webpack");
+
+webpack({
+  // [Configuration Object](/configuration/)
+}, (err, [stats](#stats-object)) => {
+  // Done
+  console.log(stats);
+});
+```
+
+**Note** that you can provide the `webpack` function with an array of configurations:
+
+``` js-with-links
+webpack([
+  { /* Configuration Object */ },
+  { /* Configuration Object */ },
+  { /* Configuration Object */ }
+], (err, [stats](#stats-object)) => {
+  // ...
+});
+```
+
+T> The `err` object **will not** include compilation errors and those must be handled separately using `stats.hasErrors()` which will be covered in detail in [Error Handling](#error-handling) section of this guide. The `err` object will only contain webpack-related issues, such as misconfiguration, etc.
```

Comment:
You can have a mini-example here (the most common use case) and still have that more extensive one below for completeness.